examples/example.c

```diff
@@ -4,18 +4,33 @@
  */
 
 #include "example_host.h"
+#ifdef STATICALLY_LINKED_EXAMPLE
 #include "example_vm/example_vm.h"
+#endif
 
 #include <evmc/helpers.h>
+#include <evmc/loader.h>
 
 #include <inttypes.h>
 #include <stdio.h>
 
 int main()
 {
+#ifdef STATICALLY_LINKED_EXAMPLE
     struct evmc_instance* vm = evmc_create_example_vm();
+    if (!vm)
+        return -1;
     if (!evmc_is_abi_compatible(vm))
         return -1;
+#else
+    enum evmc_loader_error_code error_code;
+    struct evmc_instance *vm = evmc_load_and_configure("example-vm.so", &error_code);
```

Comment:
@chfast can we extend the loader to check for dll extensions (`.so`, `.dylib`, `.dll`) so we can avoid declaring them in the load string?

Reply:
Or perhaps pass the path in `argv[1]` here.

Reply:
`argv[1]` sounds good.